creator-node/src/diskManager.ts

```diff
@@ -561,3 +568,100 @@ export async function sweepSubdirectoriesInFiles(
   // keep calling this function recursively without an await so the original function scope can close
   if (redoJob) return sweepSubdirectoriesInFiles()
 }
+
+/**
+ * Some files have been saved outside the expected `storagePath` location
+ * This job:
+ *  - fetches every file DB record that is not in expected `storagePath`
+ *  - sequentially fetches each file from network and saves to expected `storagePath`, and corrects DB record
+ *  - fetches from network since all files in this scenario are not stored on mounted volume, and so will be cleared on server restart
+ *  - on failure to fetch and store, marks DB record as skipped for later processing
+ */
+export async function fixMisplacedFiles(): Promise<void> {
+  const logger: any = createChildLogger(genericLogger, {
+    function: 'fixMisplacedFiles'
+  })
+
+  // Wrap in a try-catch to retry
+  const startTime = getStartTime()
+  try {
+    const misplacedFileRecords = await models.sequelize.query(
+      `select * from "Files" where "storagePath" not like '${getConfigStoragePath()}%';`
+    )
+    const misplacedFileRecordCount = misplacedFileRecords.length
+    logInfoWithDuration(
+      { logger, startTime },
+      `misplacedFileRecordCount = ${misplacedFileRecordCount}`
+    )
+
+    // Set in config for quick investigation
+    config.set('misplacedFileRecordCount', misplacedFileRecordCount)
+
+    const transaction = await models.sequelize.transaction()
+
+    // Process sequentially to minimize load since this is not time-sensitive
+    let lastJobSuccessfulFixMisplacedFileCount = 0
+    let lastJobFailedFixMisplacedFileCount = 0
+    for await (const fileRecord of misplacedFileRecords) {
```

Comment:
I'm curious why this is for await? afaik if you use for await...of over a sync iterable like `misplacedFileRecords` then it effectively does the same thing as for...of